packages/messaging/src/models/iid-model.ts

```diff
@@ -123,19 +172,23 @@ export default class IIDModel {
       body: fcmUnsubscribeBody
     };
 
-    const fetchResponse = await fetch(
+    return fetch(
       FCMDetails.ENDPOINT + '/fcm/connect/unsubscribe',
       unsubscribeOptions
-    );
-
-    if (!fetchResponse.ok) {
-      const fcmTokenResponse = await fetchResponse.json();
-      const message = fcmTokenResponse['error']['message'];
-      throw this.errorFactory_.create(Errors.codes.TOKEN_UNSUBSCRIBE_FAILED, {
-        message: message
-      });
-    }
-
-    return;
+    ).then(fetchResponse => {
+      if (!fetchResponse.ok) {
+        return fetchResponse.json().then(fcmTokenResponse => {
```

Comment:
This should probably have a catch added to it - sometime FCM returns HTML and `.json()` will fail on that case.

Alternative would be to parse as text, then attempt parse as JSON and if the parsing to JSON throws, we can just return the text.

Reply:
added error handling for all such cases in iid-model + test coverage.